dbms/src/Flash/Mpp/MPPTunnel.cpp

```diff
@@ -124,59 +109,46 @@ void MPPTunnel::close(const String & reason)
             cv_for_status_changed.notify_all();
             return;
         case TunnelStatus::Connected:
+        case TunnelStatus::WaitingForSenderFinish:
```

Comment:
I think `WaitingForSenderFinish` is no use. Maybe we should remove it.

Reply:
Looks like you are right, but let's remove it in a seperated pr.